ambry-api/src/main/java/com.github.ambry/clustermap/PartitionId.java

```diff
@@ -60,4 +60,9 @@
    * @return Strictly numerical string representation of the {@code PartitionId}.
    */
   public abstract String toPathString();
+
+  /**
+   * @return the partition class that this partition belongs to
```

Comment:
mention that the situations when this could be null

Reply:
Actually I see that it is non-optional when reading from json. Are we going to update helix and the static clustermaps before rolling out this change?